src/hardware/serial_terminal.rs

```diff
@@ -1,5 +1,56 @@
 use super::UsbBus;
+use crate::hardware::flash::FlashSettings;
 use core::fmt::Write;
+use miniconf::JsonCoreSlash;
+
+struct Context {
+    output: OutputBuffer,
+    flash: FlashSettings,
+}
+
+const ROOT_MENU: menu::Menu<Context> = menu::Menu {
+    label: "root",
+    items: &[
+        &menu::Item {
+            command: "read",
+            help: Some("Read a property from the device:
+
+Available Properties:
```

Comment:
Let's have a command `list` that replaces this help text as much as possible (same below for `write`). It can list the paths and print the doctring of the settings struct.